File: lab01/ejer03.py
def mostrar_menu():
    print("\n=== Menu del Cajero Automatico ===")
    print("1. Consultar Saldo")
    print("2. Depositar Dinero")
    print("3. Retirar Dinero")
    print("4. Salir")

def consultar_saldo(saldo):
    print(f"Su saldo actual es: S/. {saldo:.2f}")

def depositar(saldo):
    try:
        monto = float(input("Ingrese el monto a depositar: "))
        if monto <= 0:
            print("El monto debe ser mayor a cero.")
        else:
            saldo += monto
            print(f"Se ha depositado S/. {monto:.2f}. Nuevo saldo: S/. {saldo:.2f}")
    except ValueError:
        print("Entrada invalida. Ingrese un numero valido.")
    return saldo

def retirar(saldo):
    try:
        monto = float(input("Ingrese el monto a retirar: "))
        if monto <= 0:
            print("El monto debe ser mayor a cero.")
        elif monto > saldo:
            print("Fondos insuficientes. No se puede completar la operacion.")
        else:
            saldo -= monto
            print(f"Se ha retirado S/. {monto:.2f}. Nuevo saldo: S/. {saldo:.2f}")
    except ValueError:
        print("Entrada invalida. Ingrese un numero valido.")
    return saldo

def main():
    saldo = 1000.0
    while True:
        mostrar_menu()
        opcion = input("Seleccione una opcion (1-4): ")

        if opcion == "1":
            consultar_saldo(saldo)
        elif opcion == "2":
            saldo = depositar(saldo)
        elif opcion == "3":
            saldo = retirar(saldo)
        elif opcion == "4":
            print("Gracias por usar el cajero. Hasta luego.")
            break
        else:
            print("Opcion invalida. Intente nuevamente.")

if __name__ == "__main__":
    main()

File: lab02/src/main/java/com/miapp/ejer03.java
package lab02.src.main.java.com.miapp;

import java.util.Scanner;

public class ejer03 {
    public static void mostrarMenu() {
        System.out.println("\n=== Menú del Cajero Automático ===");
        System.out.println("1. Consultar Saldo");
        System.out.println("2. Depositar Dinero");
        System.out.println("3. Retirar Dinero");
        System.out.println("4. Salir");
    }

    public static void consultarSaldo(double saldo) {
        System.out.printf("Su saldo actual es: S/. %.2f%n", saldo);
    }

    public static double depositar(double saldo, double monto) {
        if (monto <= 0) {
            throw new IllegalArgumentException("El monto debe ser mayor a cero.");
        }
        return saldo + monto;
    }

    public static double retirar(double saldo, double monto) {
        if (monto <= 0) {
            throw new IllegalArgumentException("El monto debe ser mayor a cero.");
        } else if (monto > saldo) {
            throw new IllegalArgumentException("Fondos insuficientes.");
        }
        return saldo - monto;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double saldo = 1000.0;

        while (true) {
            mostrarMenu();
            System.out.print("Seleccione una opción (1-4): ");
            String opcion = scanner.nextLine();

            switch (opcion) {
                case "1":
                    consultarSaldo(saldo);
                    break;
                case "2":
                    try {
                        System.out.print("Ingrese el monto a depositar: ");
                        double montoDeposito = Double.parseDouble(scanner.nextLine());
                        saldo = depositar(saldo, montoDeposito);
                        System.out.printf("Nuevo saldo: S/. %.2f%n", saldo);
                    } catch (NumberFormatException e) {
                        System.out.println("Entrada inválida.");
                    } catch (IllegalArgumentException e) {
                        System.out.println(e.getMessage());
                    }
                    break;
                case "3":
                    try {
                        System.out.print("Ingrese el monto a retirar: ");
                        double montoRetiro = Double.parseDouble(scanner.nextLine());
                        saldo = retirar(saldo, montoRetiro);
                        System.out.printf("Nuevo saldo: S/. %.2f%n", saldo);
                    } catch (NumberFormatException e) {
                        System.out.println("Entrada inválida.");
                    } catch (IllegalArgumentException e) {
                        System.out.println(e.getMessage());
                    }
                    break;
                case "4":
                    System.out.println("Gracias por usar el cajero. Hasta luego.");
                    return;
                default:
                    System.out.println("Opción inválida. Intente nuevamente.");
            }
        }
    }
}

